tasm-tour/src/components/ExhibitPage/ExhibitPage.js

```diff
@@ -14,20 +16,26 @@ export default function ExhibitPage({ exhibitID }) {
 
   useEffect(() => {
     const fetchData = async () => {
+      setError(null);
+      setLoading(true);
+      setExhibit(null);
       try {
         const db = getFirestore(app);
-        const exhibitRef = doc(db, 'exhibits', exhibitID);
-        const exhibitSnapshot = await getDoc(exhibitRef);
-
-        if (exhibitSnapshot.exists()) {
+        const queryResults = query(collection(db, 'exhibits'), where('fourDigitCode', '==', exhibitID));
+        const exhibitResults = await getDocs(queryResults);
+        if (!exhibitResults.empty) {
+          const exhibitSnapshot = exhibitResults.docs[0];
           setExhibit(exhibitSnapshot.data());
+          if (exhibitResults.size > 1) {
+            console.warn("Multiple exhibits found with the same ID: " + exhibitID);
+          }
```

Comment:
Should this logic be before calling setExhibit?
Perhaps after checking if results is empty?

Reply:
Don't think it matters where you stick it. works the same either way. but yes it would technically be more readable